crypto/keys/sr25519/sr25519_test.go

```diff
@@ -0,0 +1,224 @@
+			},
+			true,
+		},
+		{
+			"different types",
+			sr25519PrivKey,
+			secp256k1.GenPrivKey(),
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.msg, func(t *testing.T) {
+			eq := tc.privKey.Equals(tc.other)
+			require.Equal(t, eq, tc.expectEq)
+		})
+	}
+}
+
+func TestMarshalAmino(t *testing.T) {
+	aminoCdc := codec.NewLegacyAmino()
+	privKey := sr25519.GenPrivKey()
+	pubKey := privKey.PubKey().(*sr25519.PubKey)
+
+	testCases := []struct {
+		desc      string
+		msg       codec.AminoMarshaler
+		typ       interface{}
+		expBinary []byte
+		expJSON   string
+	}{
+		{
+			"sr25519 private key",
+			privKey,
+			&sr25519.PrivKey{},
+			append([]byte{sr25519.PrivKeySize}, privKey.Bytes()...), // Length-prefixed.
+			"\"" + base64.StdEncoding.EncodeToString(privKey.Bytes()) + "\"",
+		},
+		{
+			"sr25519 public key",
+			pubKey,
+			&sr25519.PubKey{},
+			append([]byte{sr25519.PubKeySize}, pubKey.Bytes()...), // Length-prefixed.
+			"\"" + base64.StdEncoding.EncodeToString(pubKey.Bytes()) + "\"",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			// Do a round trip of encoding/decoding binary.
+			bz, err := aminoCdc.MarshalBinaryBare(tc.msg)
+			require.NoError(t, err)
+			require.Equal(t, tc.expBinary, bz)
+
+			err = aminoCdc.UnmarshalBinaryBare(bz, tc.typ)
+			require.NoError(t, err)
+
+			require.Equal(t, tc.msg, tc.typ)
+
+			// Do a round trip of encoding/decoding JSON.
+			bz, err = aminoCdc.MarshalJSON(tc.msg)
+			require.NoError(t, err)
+			require.Equal(t, tc.expJSON, string(bz))
+
+			err = aminoCdc.UnmarshalJSON(bz, tc.typ)
+			require.NoError(t, err)
+
+			require.Equal(t, tc.msg, tc.typ)
+		})
+	}
+}
+
+func TestMarshalAmino_BackwardsCompatibility(t *testing.T) {
+	aminoCdc := codec.NewLegacyAmino()
+	// Create Tendermint keys.
+	tmPrivKey := tmsr25519.GenPrivKey()
+	tmPubKey := tmPrivKey.PubKey()
+	// Create our own keys, with the same private key as Tendermint's.
+	privKey := &sr25519.PrivKey{MiniSecretKey: []byte(tmPrivKey)}
+	pubKey := privKey.PubKey().(*sr25519.PubKey)
+
+	testCases := []struct {
+		desc      string
+		tmKey     interface{}
+		ourKey    interface{}
+		marshalFn func(o interface{}) ([]byte, error)
+	}{
+		{
+			"sr25519 private key, binary",
+			tmPrivKey,
+			privKey,
+			aminoCdc.MarshalBinaryBare,
+		},
+		{
+			"sr25519 private key, JSON",
+			tmPrivKey,
+			privKey,
+			aminoCdc.MarshalJSON,
+		},
+		{
+			"sr25519 public key, binary",
+			tmPubKey,
+			pubKey,
+			aminoCdc.MarshalBinaryBare,
+		},
+		{
+			"sr25519 public key, JSON",
+			tmPubKey,
+			pubKey,
+			aminoCdc.MarshalJSON,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			// Make sure Amino encoding override is not breaking backwards compatibility.
+			bz1, err := tc.marshalFn(tc.tmKey)
+			require.NoError(t, err)
+			bz2, err := tc.marshalFn(tc.ourKey)
+			require.NoError(t, err)
+			require.Equal(t, bz1, bz2)
+		})
+	}
+}
```

Comment:
Please add a test for `cryptotypes.PubKey` interface marshaling. Here is an example: https://github.com/cosmos/cosmos-sdk/blob/master/crypto/keys/secp256r1/pubkey_internal_test.go#L96